discovery-provider/src/utils/prometheus_metric.py

```diff
@@ -7,6 +8,28 @@
 logger = logging.getLogger(__name__)
 
 
+def save_duration_metric(metric_group):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            metric = PrometheusMetric(
+                f"{metric_group}_duration_seconds",
+                f"How long a {metric_group} took to complete",
+                ("func_name", "success"),
+            )
+            try:
+                result = func(*args, **kwargs)
+                metric.save_time({"func_name": func.__name__, "success": True})
```

Comment:
func.__name__ is the name of the @celery.task?

Reply:
```
@celery.task(name="index_trending", bind=True)
@save_duration_metric(metric_group="celery_task")
def index_trending_task(self):
```

In the above example, `func.__name__` will be `index_trending_task`.

This is because:

* decorators are applied from the bottom up,
* the `@celery.task` must be the last decorator applied,
* `@save_duration_metric` will only have access to the caller function.